src/org/rascalmpl/library/lang/rascal/tests/concrete/Parsing.rsc

```diff
@@ -62,3 +62,9 @@ test bool parsingWithAManualGrammar()
   && Tree t := parse(gr, "hello")
   && "<t>" == "hello";
   
+test bool saveAndRestoreParser() {
+  storeParsers(#start[A], |test-temp:///parsers.jar|);
+  p = loadParsers(|test-temp:///parsers.jar|);
+
+  return p(type(\start(sort("A")), ()), "a", |origin:///|) == parse(#start[A], "a", |origin:///|);
```

Comment:
nice test :) can we also save a file somewhere? and load it back in?

Reply:
I don't understand the question. The test saves the file in `|test-temp:///parsers.jar|` and then loads it back from that file, assigning the parser function into `g`. Then it tests if that recovered parser function behaves the same as a freshly generated parser function.

Reply:
but this code already has a `#start[A]`  in there. So what if we load a grammar that isn't reified in the code?

Reply:
Excellent question. I've coded that scenarios in the @examples of the documentation, because there I can freely start a fresh REPL that has no declaration memory. I can not do that here unfortunately. Nevertheless, yes that scenarios works 100%. It's a bit weird, because you do get parse trees for types that have never been declared. But, it works.

Reply:
The reason that it works is that the `parsers` function and the `storeParsers/loadParsers` function generate functions that take only the _type_ as parameter and ignore the grammar part of the type values. The grammar has already been used by the `store` function, and the start type is only used to decide which part of the grammar to activate.